util.c

```diff
@@ -570,7 +570,12 @@ int insert_image(struct image *image, struct image *sub,
 		image_error(image, "open %s: %s", infile, strerror(errno));
 		goto out;
 	}
-	ret = map_file_extents(image, infile, in_fd, size, &extents, &extent_count);
+	if (byte == 0) {
+		ret = map_file_extents(image, infile, in_fd, size, &extents, &extent_count);
+	} else {
+		/* Don't want to fill holes with non-0 bytes */
+		ret = whole_file_exent(size, &extents, &extent_count);
+	}
```

Comment:
So I've been thinking about this some more. You could also change the last argument of the `write_bytes()` call in the loop below to `0`. This makes the assumption that 'holes' are always zero. I think that's save to assume.

This would save some disk space. Not sure how much. That depends on the input images.